PythonRegression/tests/features/machine1/1_api_tests.feature

```diff
@@ -38,56 +53,174 @@ Feature: Test API calls on Machine 1
-		Given "getTransactionsToApprove" is called on "nodeA"
+		Given "getTransactionsToApprove" is called on "nodeA" with:
+		|keys       |values                 |type           |
+        |depth      |3                      |int            |
+
 		Then a response with the following is returned: 
 		|keys								|
-		|trunkTransaction					|
 		|branchTransaction					|
 		|duration							|
-		
+		|trunkTransaction					|
+
+
+    Scenario: CheckConsistency is called
+        Given "checkConsistency" is called on "nodeA" with:
+        |keys           |values             |type           |
+        |tails          |TEST_HASH          |staticList     |
+
+        Then a response with the following is returned:
+        |keys                               |
+        |duration                           |
+        |info                               |
+        |state                              |
+
+
+    #Values can be found in util/static_vals.py
+    Scenario: GetInclusionStates is called
+	    Given "getInclusionStates" is called on "nodeA" with:
+        |keys           |values             |type               |
+        |transactions   |TEST_HASH          |staticList         |
+        |tips           |TEST_TIP_LIST      |staticValue        |
+
+        Then a response with the following is returned:
+        |keys                               |
+        |duration                           |
+        |states                             |
+
+
+    #Address can be found in util/static_vals.py
+    Scenario: GetBalances is called
+        Given "getBalances" is called on "nodeA" with:
+        |keys       |values                 |type               |
+        |addresses  |TEST_EMPTY_ADDRESS     |staticList         |
+        |threshold  |100                    |int                |
+
+
+        Then a response with the following is returned:
+        |keys                               |
+        |balances                           |
+        |duration                           |
+        |milestoneIndex                     |
+        |references                         |
+
+
+    Scenario: Interrupt attach to tangle
+        Given "attachToTangle" is called on one thread in "nodeA" with:
+        |keys                   |values                         |type           |
+        |trytes                 |EMPTY_TRANSACTION_TRYTES       |staticList     |
+        |trunk_transaction      |TEST_HASH                      |staticValue    |
+        |branch_transaction     |TEST_HASH                      |staticValue    |
+        |min_weight_magnitude   |50                             |int            |
+
+        And "interruptAttachingToTangle" is called on one thread in "nodeA" with:
+        |keys                   |values                         |type           |
+
+        When all api call threads are joined
+        #The only active thread should be the main testing thread
+        Then while checking for running threads, there should be "1" running
```

Comment:
I doubt that this actually tests whether `interruptAttachingToTangle` really works...
I think you can put this code in a git stash (or save it somehow).

You can open a separate issue for it.

Reply:
The test now ensures that the interrupt causes `attachToTangle` to return an empty tryte list